src/vmq_queue.erl

```diff
@@ -296,6 +275,13 @@ drain(Event, State) ->
     lager:error("got unknown event in drain state ~p", [Event]),
     {next_state, drain, State}.
 
+drain({enqueue_many, Msgs}, _From, #state{drain_over_timer=TRef} =  State) ->
+    %% if multiple queues trigger queue migration at the 'same' time
+    %% so we've to queue those message otherwise they would be lost.
+    gen_fsm:cancel_timer(TRef),
+    gen_fsm:send_event(self(), drain_start),
```

Comment:
Oh, so we then cancel the timer and tell ourselves to go back to state `drain_start`. Couldn't we just explicitly go to that state in line 284?